crates/edr_provider/src/data.rs

```diff
@@ -2399,6 +2400,39 @@ where
             },
         )?
     }
+
+    /// Filters out transactions with unsupported types and returns the
+    /// remaining transactions, if skipping is allowed. Otherwise returns
+    /// an error.
+    fn filter_unsupported_transaction_types(
+        &self,
+        transactions: &[ChainSpecT::Transaction],
+        transaction_hash: &B256,
+    ) -> Result<Vec<ChainSpecT::Transaction>, ProviderError<ChainSpecT>> {
+        transactions
+            .iter()
+            .filter_map(|transaction| {
+                if transaction.is_executable_transaction() {
+                    Some(Ok(transaction.clone()))
```

Comment:
Cloning here is a slight performance regression compared to main, but I don't think it matters compared to the cost of executing the transactions.

Reply:
On main we were [calling `to_vec`](https://github.com/NomicFoundation/edr/blob/main/crates/edr_provider/src/data.rs#L1985) which would also have cloned the transactions.

https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec

Reply:
Only for locally mined blocks which will typically have a lot fewer transactions than remote blocks. The remote path [takes](https://github.com/NomicFoundation/edr/blob/c8fbf3b2a6ca4f838d408a47536668a0c413c256/crates/edr_provider/src/data.rs#L2015) the transactions from the remote block.

Reply:
Gotcha! 👍